wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsnext/model/ChePlugin.java

```diff
@@ -14,78 +14,20 @@
 import java.util.List;
 import java.util.Objects;
 
-public class ChePlugin {
+/** Represents Che plugin in sidecar-powered workspace. */
+public class ChePlugin extends PluginBase {
+  private List<EditorCompatibility> editors = new ArrayList<>();
```

Comment:
I'm not sure that I fully understand all last changes in WsNext Skeleton, and it's just a note to consider: in `editors` field I would expect to see a list of CheEditors objects, or list of CheEditors identifiers.

Reply:
editor field contains a list of editors supported by a plugin with identifiers in a field `name` and other plugins that needed for correct integration of the editor and the plugin in a field `plugins`. All of that is incorporated in EditorCompatibility object in this PR.
Does this clarify the idea to you?

Reply:
Yeah, thanks for the explanation.

Does it mean that different CheEditors can be used at the same time? It's OK just to clarify.

As far as I understand `ChePlugin` depends on `EditorCompatibility`s, `EditorCompatibility` depends on `ChePlugin`s. Are circular dependencies are handled correctly?

Reply:
ChePlugin shows that different plugins might be used with it, but Che should ensure that just one is in use.
Circular dependencies are not handled because compatibility is not implemented and I'm not sure we need this verification on this step where parts are still moving rapidly and things that are here now might gone tomorrow.